Issue 1130 - native uris should be valid

philkcw opened the pull request:
### Summary
Fixes https://github.com/doorkeeper-gem/doorkeeper/issues/1130

philkcw commented:
Sigh.  How can code coverage decrease when I added one line and one test?

ThibG commented:
This is missing changes to the test file:
https://github.com/doorkeeper-gem/doorkeeper/pull/1060/files#diff-b2085d14cb8374956712da8c8d55e8ef

nbulaj commented:
The tests are red. Also could you please squash all the commits to a single one once it finished? Thanks!

philkcw commented:
@ryansch I tried to add in your other change but it broke your tests and my tests as well, so I reverted back to the one-line change that I'm most interested in. 😄

philkcw commented:
@ryansch @thibaudgg I'll take another look at the changes you mentioned this weekend.

thibaudgg commented:
@philkcw I think you meant @ThibG and not me :)

ryansch commented:
Ha! Fair enough!

Gargron commented:
I just want to tell you both good luck. We're all counting on you.

philkcw commented:
@ryansch I added in your change.  @ThibG I added in your tests.  Maybe I should have cherry-picked instead?  @nbulaj the build passes and I've squashed.  Any more comments?

philkcw commented on spec/models/doorkeeper/application_spec.rb:
```diff
@@ -266,6 +266,23 @@ module Doorkeeper
         it { expect(subject).to eq(false) }
       end
+
+      context 'when the application does not support confidentiality' do
```
I added this to get the code coverage to pass.

nbulaj commented:
Looks good to me too, thanks!
